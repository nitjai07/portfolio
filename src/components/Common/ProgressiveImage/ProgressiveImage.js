import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProgressiveImage extends Component {
  state = {
    piTinyImageSource: this.props.piTinyImageSource,
    imgLoaded: false
  };

  componentDidMount() {
    this.PreloadImage(this.props.piMainImageSource);
  }

  PreloadImage = src => {
    const img = new Image();
    img.src = src;
    img.onload = () => {
      this.setState((prevState, props) => ({
        piTinyImageSource: props.piMainImageSource,
        imgLoaded: true
      }));
    };
  };

  render() {
    return (
      <img
        style={this.state.imgLoaded ? {} : { filter: this.props.piFilterEffect }}
        height={this.props.piHeight}
        width={this.props.piWidth}
        className={`${this.props.customClass}`}
        src={this.state.piTinyImageSource}
        alt={this.props.piAlt}
      />
    );
  }
}

ProgressiveImage.propTypes = {
  piTinyImageSource: PropTypes.string.isRequired,
  piMainImageSource: PropTypes.string.isRequired,
  customClass: PropTypes.string,
  piAlt: PropTypes.string,
  piHeight: PropTypes.string,
  piWidth: PropTypes.string,
  piFilterEffect: PropTypes.string
};
ProgressiveImage.defaultProps = {
  customClass: '',
  piAlt: '',
  piHeight: '100%',
  piWidth: '100%',
  piFilterEffect: ''
};

export default ProgressiveImage;
