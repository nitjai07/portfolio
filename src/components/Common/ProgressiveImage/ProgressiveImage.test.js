import React from 'react';
import { shallow } from 'enzyme';
import ProgressiveImage from './ProgressiveImage';
import meImg from '../../../assets/images/me_1.jpg';
import meImLow from '../../../assets/images/me_1_low.jpg';

test('ProgressiveImage renders correctly', () => {
  const component = shallow(<ProgressiveImage piTinyImageSource={meImLow} piMainImageSource={meImg} />);
  expect(component).toMatchSnapshot();
});
