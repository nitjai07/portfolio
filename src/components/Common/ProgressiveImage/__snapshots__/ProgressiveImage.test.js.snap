// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <img
    alt=""
    className=""
    height="100%"
    src="me_1_low.jpg"
    style={
        Object {
            "filter": "",
          }
    }
    width="100%"
/>,
  "nodes": Array [
    <img
      alt=""
      className=""
      height="100%"
      src="me_1_low.jpg"
      style={
            Object {
                  "filter": "",
                }
      }
      width="100%"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <ProgressiveImage
      customClass=""
      piAlt=""
      piFilterEffect=""
      piHeight="100%"
      piMainImageSource="me_1.jpg"
      piTinyImageSource="me_1_low.jpg"
      piWidth="100%"
/>,
    "_forcedUpdate": false,
    "_instance": ProgressiveImage {
      "PreloadImage": [Function],
      "context": Object {},
      "props": Object {
        "customClass": "",
        "piAlt": "",
        "piFilterEffect": "",
        "piHeight": "100%",
        "piMainImageSource": "me_1.jpg",
        "piTinyImageSource": "me_1_low.jpg",
        "piWidth": "100%",
      },
      "refs": Object {},
      "state": Object {
        "imgLoaded": false,
        "piTinyImageSource": "me_1_low.jpg",
      },
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <img
      alt=""
      className=""
      height="100%"
      src="me_1_low.jpg"
      style={
            Object {
                  "filter": "",
                }
      }
      width="100%"
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ProgressiveImage
    customClass=""
    piAlt=""
    piFilterEffect=""
    piHeight="100%"
    piMainImageSource="me_1.jpg"
    piTinyImageSource="me_1_low.jpg"
    piWidth="100%"
/>,
}
`;
